### PR TITLE
Refactor `MagmaLaw.op`

### LINKED ISSUE #525: LOW PRIORITY: Give Magma.op and MagmaLaw.op less similar names
See the discussion at https://leanprover.zulipchat.com/#narrow/stream/458659-Equational/topic/Magma.2Eop.20vs.20MagmaLaw.2Eop .  There is no immediate conflict, so this is not an urgent issue, but it may be good to propose less similar names for these rather different operations, and refactor at some point.

### PR DESCRIPTION
**Pull request by pitmonticone**
null

Closes #525

### CODE REVIEW

File: equational_theories/MagmaOp.lean [hunk 1]
```diff
@@ -44,27 +44,27 @@ theorem models.Op {α} {G : Type} [Magma G] {w₁ w₂ : FreeMagma α} (h : G 
 
 namespace Law.MagmaLaw
 
-def op {α} (l : MagmaLaw α) : MagmaLaw α := { lhs := l.lhs.op, rhs := l.rhs.op}
+def dual {α} (l : MagmaLaw α) : MagmaLaw α := { lhs := l.lhs.op, rhs := l.rhs.op}
```
Comment (goens) [inline equational_theories/MagmaOp.lean:47]: I think these should align (i.e. the `op`s on `l.lhs` and `l.rhs`, `FreeMagma`s should alse become `dual`s

Comment (pitmonticone) [reply equational_theories/MagmaOp.lean:47]: Agree. Will do it later. Thanks for the comment!

Comment (pitmonticone) [reply equational_theories/MagmaOp.lean]: @goens Done, thanks. Can you review it again now?

**Review by goens — APPROVED** [review]
LGTM, thanks!

**Pull request merged**